Update forecast ensembles

nickreich opened the pull request.

seabbs commented on sessions/forecast-ensembles.qmd:
```diff
@@ -6,40 +6,44 @@ bibliography: ../nfidd.bib
-The aim of this session is to introduce the concept of ensembles of forecasts and to evaluate the performance of ensembles of the models we explored in the [forecasting models](forecasting-models) session.
-
-::: {.callout-note collapse="true"}
+The aim of this session is to introduce the concept of ensembles of forecasts and to evaluate the performance of ensembles of the models we explored in the earlier sessions.
```
Is there a reason to not include a link to the previous sessions?

seabbs commented on sessions/forecast-ensembles.qmd:
```diff
@@ -48,127 +52,230 @@ The source file of this session is located at `sessions/forecast-ensembles.qmd`.
+While `fable` is good for some fairly simple modeling, we will want and need more complexity and a more flexible representation of forecast data.
+In this section, we will start using hubverse-style data for forecast output formats.
+
+::: callout-tip
```
do you want to collapse this?

seabbs commented on sessions/forecast-ensembles.qmd:
```diff
@@ -48,127 +52,230 @@ The source file of this session is located at `sessions/forecast-ensembles.qmd`.
+A strength of the hubverse data standards is that it facilitates comparison and synthesis of models that are made with a wide variety of different modeling tools.
+If you can format your output data according to the simple standards, then your model could be part of a modeling hub.
+
+## Reformatting model output
```
this is a lot of information about data formatting perhaps it can go in a collapsed call out?

seabbs commented on sessions/forecast-ensembles.qmd:
```diff
@@ -48,127 +52,230 @@ The source file of this session is located at `sessions/forecast-ensembles.qmd`.
 This corresponds to a situation where each forecast aims to correctly determine a single target predictive distribution.
 By taking an average of all models, we aim to get a better estimate of this distribution than from the individual models.
-If we have reason to believe that some models are better than others at estimating this distribution, we can create a weighted version of this average.
+If we have reason to believe that some models are **consistently** better than others at estimating this distribution, we could try to create a weighted version of this average.
```
I don't think this is quite correct. We can always create a weighted version its just that it might not be any good. Suggest an edit.

seabbs commented on sessions/forecast-ensembles.qmd:
````diff
@@ -425,467 +474,231 @@ We combine these new ensembles with our previous ensembles in order to make visu
+To facilitate visual and quantitative comparisons with the earlier ensemble forecasts, we convert the new ensemble forecasts to quantiles and then plot them.
+
+```{r, fig.height = 12}
+all_forecasts <- bind_rows(
````
seems like an unsafe name?

seabbs commented on sessions/forecast-ensembles.qmd:
```diff
@@ -425,467 +474,231 @@ We combine these new ensembles with our previous ensembles in order to make visu
-How do the weighted ensembles compare to the simple ensembles?
+### Solution
 
+The LOP ensembles tend to be a bit wider than the other ensemble forecasts, especially at times when the forecasts disagree more.
```
Is this solution linked to the question? I think both are useful but probably need an edit to align.

seabbs commented on sessions/forecast-ensembles.qmd:
```diff
@@ -425,467 +474,231 @@ We combine these new ensembles with our previous ensembles in order to make visu
-Think about the reliability and validity of the resulting ensemble forecast. 
-- Would it change how you approach communicating the forecast?
+-   Given the performance by forecast date shown by the different models, would you expect a weighted ensemble to perform better than the unweighted ensemble?
+-   
```
lose -

seabbs commented on sessions/forecast-ensembles.qmd:
```diff
@@ -425,467 +474,231 @@ We combine these new ensembles with our previous ensembles in order to make visu
+-   Given the performance by forecast date shown by the different models, would you expect a weighted ensemble to perform better than the unweighted ensemble?
+-   
 
 ## Methods in the real world
```
link to some methods/tools that do support weighting.

seabbs reviewed (approved):
I think it's a shame to tell not show with the weighted ensembles and the current data formatting is a bit tangential and long but ultimaetely its fine. Other than that the content looks good and I like the use of the howerton figure. Some small suggestions

seabbs commented on sessions/forecast-ensembles.qmd:
```diff
@@ -48,127 +52,230 @@ The source file of this session is located at `sessions/forecast-ensembles.qmd`.
-library("tidyr")
 library("ggplot2")
-library("scoringutils")
-library("qrensemble")
```
and also here. Scoringutils isn't used directly and so shouldn't need to be loaded or installed.

seabbs commented:
See here for loading in the hubverse dependency URLs: https://github.com/nfidd/sismid/pull/59

Given the only difference is the hubverse packages not being installed I wonder if that is the issue (i.e. they have something going on with additional repositories etc that is causing an issue)?

nickreich commented:
So... I removed hubData (I think not needed).

hubUtils and hubEnsembles are on CRAN. And when just those two packages were included in DESCRIPTION the non-deploy actions were working.

But as soon as I added back in hubEvals (only on R-universe), the actions started failing again.

So something with the R-universe integration with CI is not working right. These [two](https://docs.r-universe.dev/install/dependencies.html#can-a-cran-package-take-on-a-dependency-from-r-universe) [links](https://github.com/r-lib/remotes/issues/618) suggested to me that we are doing the right things:

nickreich commented:
I'm curious why it's only failing on the hubverse R-universe packages. Are there other R-universe packages really loading here? or are all the other packages on CRAN?
